src/Monocle/Api/Server.hs

```diff
@@ -934,24 +967,37 @@ handleLoggedIn err codeM stateM = do
             (from oauthCode)
       now <- liftIO Monocle.Prelude.getCurrentTime
       let idToken = O.idToken tokens
-          expiry = addUTCTime (24 * 3600) now
+          dayS = 24 * 3600
+          expiry = addUTCTime dayS now
           userId = aUserId oidcEnv idToken
           mUidMap = getIdents config $ "AuthProviderUID:" <> userId
+          authenticatedUser = AUser mUidMap userId (truncate $ nominalDiffTimeToSeconds $ utcTimeToPOSIXSeconds expiry)
+          jwtCfg = localJWTSettings aOIDC
       logInfo "OIDCProviderTokenRequested" ["id" .= show @Text idToken]
-      jwtE <- liftIO $ mkJwt (localJWTSettings aOIDC) mUidMap userId (Just expiry)
-      case jwtE of
-        Right jwt -> do
+      -- Here we create the JWT Session Cookie that will be used by the browser to authenticate requests
+      mApplyCookies <-
+        liftIO $
+          acceptLogin
+            -- The nested JWT owns the expiry
+            (cookieSettings {cookieExpires = Just expiry})
+            jwtCfg
+            authenticatedUser
+      -- Here we create a already expired JWT to be set as the Monocle Cookie. This cookie is used by web app
+      -- to extract non sensible authenticated user info
```

Comment:
Why not provide this information through the info endpoint instead? That sounds like a design issue, we shouldn't have to provide multiple cookies like that.

Reply:
By changing the way the JWT is shared between the Web APP and the API I wanted to minimize the amount of changes especially inside the Web APP, by also keeping the current Web APP behavior, which is the case with that change. However I agree there are rooms for improvement like as you said relying only on the API to get the user's info and remove the additional cookie. I'll create a new followup story.